File: Assignment-2/Assn2.java
//author: Sotheanith Sok
//Import
import java.io.*;
import java.util.Scanner;
public class Assn2{
    public static void main(String[] args) {
        try{
            //Open file
            File dir =new File("./" + args[0]);

            //Create scanner
            Scanner sc=new Scanner(dir);

            //Define regex
            String regex="^\\$\\**((0)|(([1-9][0-9]?[0-9]?)(,[0-9][0-9][0-9])*))((\\.[0-9][0-9])*)?$"; // Check for decimal digits after period.

            //Start to read, compare, and prinout 
            while(sc.hasNextLine()){
                String text=sc.nextLine();
                if(text.matches(regex)){
                    System.out.println("Matched: "+text);
                }else{
                    System.out.println("Not Matched: "+text);
                }
            }

            //Close scanner
            sc.close();
        }catch(FileNotFoundException fileNotFoundException){
            System.out.println(fileNotFoundException);
        }
        
    }
}

File: Assignment-2/Assn2.py
#author: Sotheanith Sok
#import
import sys
import re
import warnings

#supress warning
warnings.simplefilter(action='ignore', category=FutureWarning)

#Define regex
regex = r"^[$][*]*([0-9]|(([1-9][0-9]?[0-9]?)([,][0-9][0-9][0-9])*))([[.][0-9][0-9]]*)?$"

#Open file, read, compare, and print out result
with open(sys.argv[1],"r") as file:
    for line in file:
        line=line.strip('\n')
        matches=re.match(regex,line)
        if(matches):
            print("Matched: ",line)
        else:
            print("Not Matched: ",line)
    
